### ISSUE TITLE
New SoftMax has bug in gradient

### ISSUE THREAD

**Issue by wzzhu** [gorgonia/gorgonia#487]
go test -v -run TestOld_NewSoftmax
=== RUN   TestOld_NewSoftmax
    op_softmax_test.go:160: 
        [0.17949255984499132  0.17949255984499132  0.21923270766390177  0.17949255984499132   0.2422896128011244]
        [0.17949255984499132  0.17949255984499132  0.21923270766390174  0.17949255984499132  0.24228961280112435]
    op_softmax_test.go:161: 
        **[ -3.985535453776364e-17   -3.985535453776364e-17   -4.867943995987595e-17   -3.985535453776364e-17  -5.3799101351864477e-17]
        [ -3.067465498484267e-17   -3.067465498484267e-17  -3.7466108204098436e-17   -3.067465498484267e-17   -4.140645320064502e-17]**

--- PASS: TestOld_NewSoftmax (0.00s)

Though the test says it passes, the old/new gradients are different (bold lines)

The failure of new softmax op may make the whole network unusable since softmax is widely used in neural network training.

**Comment by wzzhu** [comment]
Looks like it is the floating point resolution problem, close it for it. The gradient is 0 for all.
